### ISSUE TITLE
Incorrect line/col number when contentVersion is mismatched in linked templates

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode-azurearmtools#1144]
The problem is that lineNumber/columnNumber are generally considered to be relative to the linked/nested template's content, not the main template's content, but in this case I'm getting the location of contentVersion in the main template.  Since I'm assuming this is relative to the template's contents, I end up with the wrong location for the error.  You either need to use relative locations or have a flag or something.

The other possibility - I'm not positive the SourceType should be linkedTemplate, since this is not occuring inside the linked template itself.  Thoughts?

![image](https://user-images.githubusercontent.com/6913354/105260555-d4238980-5b42-11eb-9251-62d685b8065c.png)

**Comment by jorgecotillo** [comment]
Can you please add your sample template so I can repro?

**Comment by StephenWeatherford** [comment]
main.json:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {},
    "functions": [],
    "variables": {},
    "resources": [
        {
            "name": "linkedDeployment1",
            "type": "Microsoft.Resources/deployments",
            "apiVersion": "2020-10-01",
            "properties": {
                "mode": "Incremental",
                "templateLink": {
                    "relativePath": "child.json",
                    "contentVersion": "2.0.0.0"
                },
                "parameters": {
                }
            }
        }
    ],
    "outputs": {}
}
```
child.json:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "3.0.0.0",
    "parameters": {},
    "functions": [],
    "variables": {},
    "resources": [
        {
            "name": "[uniqueString(resourceGroup().id, 'storageaccount1')]",
            "type": "Microsoft.Storage/storageAccounts",
            "apiVersion": "2019-06-01",
            "tags": {
                "displayName": "storageaccount1"
            },
            "location": "[resourceGroup().location]",
            "kind": "StorageV2",
            "sku": {
                "name": "Premium_LRS",
                "tier": "Premium"
            }
        }
    ],
    "outputs": {}
}
```
main.parameters.json:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
    }
}
```
Expecting an error since 2.0.0.0 doesn't match 3.0.0.0.

**Comment by StephenWeatherford** [comment]
This still repros with 1.0.31

**Comment by StephenWeatherford** [comment]
Fixed by #1241
